### PR TITLE
[ETCM-126] add eth_pendingTransactions

### PR DESCRIPTION
**Pull request by lemastero**
# Description

Add support for Ethereum JSON-RPC endpoint [eth_pendingTransactions](https://github.com/etclabscore/ethereum-json-rpc-specification/blob/master/openrpc.json#L850-L859)

# Proposed Solution

# Important Changes Introduced

# Testing

### CODE REVIEW

**Comment by mmrozek** [comment]
I think it works the same as `qa_getPendingTransactions`. So probably we could remove the old method

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerSpec.scala [hunk 1]
```diff
@@ -1901,6 +1902,71 @@ class JsonRpcControllerSpec
...
+      ),
+      Some(JInt(1))
+    )
+
+    val response: JsonRpcResponse = jsonRpcController.handleRequest(request).futureValue
+
+    response should haveResult(JArray(List()))
+  }
+
+  it should "mempool is empty" in new TestSetup {
```
Comment (mmrozek) [inline src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerSpec.scala:1926]: Mempool has transactions here

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerSpec.scala [hunk 2]
```diff
@@ -1901,6 +1902,71 @@ class JsonRpcControllerSpec
     response should haveObjectResult("transactions" -> JArray(expectedTxs.toList))
   }
 
+  "request pending transactions and return valid response" should "mempool is empty" in new TestSetup {
```
Comment (mmrozek) [inline src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerSpec.scala:1905]: I think we should start to split this file. It is already too big. Maybe sth like `JsonRpcControllerHandlingTransactionsSpec`? WDYT?

File: src/main/scala/io/iohk/ethereum/jsonrpc/QAService.scala [hunk 3]
```diff
@@ -1,32 +1,25 @@
 package io.iohk.ethereum.jsonrpc
 
-import akka.actor.ActorRef
+import akka.util.ByteString
```
Comment (mmrozek) [inline src/main/scala/io/iohk/ethereum/jsonrpc/QAService.scala:3]: Minor: double import

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerSpec.scala (hunk 2 above)
Comment (lemastero) [reply src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerSpec.scala]: Yes, I highly agree. I try to do this in this PR but work it is a lot of changes and it shadows the real work done in here + somehow tests got broken. So it is better to go small steps.

I split JsonRpc tests in separate PR: https://github.com/input-output-hk/mantis/pull/734

File: src/main/scala/io/iohk/ethereum/jsonrpc/QAService.scala (hunk 3 above)
Comment (lemastero) [reply src/main/scala/io/iohk/ethereum/jsonrpc/QAService.scala:3]: Good spot. Fixed.

File: src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerSpec.scala (hunk 1 above)
Comment (lemastero) [reply src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerSpec.scala]: Good catch, thank you it gave me some good hint on why tests failed after porting from qa scope to eth 👍

**Comment by lemastero** [comment]
> I think it works the same as `qa_getPendingTransactions`. So probably we could remove the old method

Done.

File: insomnia_workspace.json [hunk 4]
```diff
@@ -961,6 +961,43 @@
       "settingFollowRedirects": "global",
       "_type": "request"
     },
+    {
```
Comment (mmrozek) [inline insomnia_workspace.json:964]: Please remove `qa_getPendingTransactions` from insomnia also

File: build.sbt [hunk 5]
```diff
@@ -146,3 +146,13 @@ addCommandAlias(
     |;benchmark:compile
     |""".stripMargin
 )
+
+// prepare PR
```
Comment (mmrozek) [inline build.sbt:150]: 👍

**Review by mmrozek — APPROVED** [review]
LGTM!

File: insomnia_workspace.json (hunk 4 above)
Comment (kapke) [reply insomnia_workspace.json:927]: And remember to notify QA once this gets merged :)

**Review by kapke — APPROVED** [review]
LGTM!

**Pull request merged**